### PR TITLE
[#274] Autoencoders unit and integration tests for each pipeline step

### LINKED ISSUE #274: (AE) unit and integration tests for each pipeline step

### PR DESCRIPTION
**Pull request by Michal-Kolomanski**
closes #274 #253

### CODE REVIEW

File: recommender/engines/base/base_pipeline.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def __init__(self, pipeline_config: dict):
             ModelEvaluationStep.__name__: self._create_model_evaluation_step(),
             ModelValidationStep.__name__: self._create_model_validation_step(),
         }
-        self.verbose = self.pipeline_config.get(VERBOSE, True)
+        self.verbose = self.pipeline_config.get(VERBOSE, False)
```
Comment (Michal-Kolomanski) [inline recommender/engines/base/base_pipeline.py:37]: But, on second consideration, shouldn't we stick with True here?

File: recommender/engines/autoencoders/inference/embedding_component.py [hunk 2]
```diff
@@ -22,11 +22,9 @@ def __init__(self):
     def __call__(self, use_cache=False, save_cache=True):
         """Performs embedding with normalization."""
 
-        print("Starting embedding...")
```
Comment (Michal-Kolomanski) [inline recommender/engines/autoencoders/inference/embedding_component.py:25]: Keeping those print statements makes more sense with the new commands [GH](https://github.com/cyfronet-fid/recommender-system/blob/54bd5a994b72705c78c6a10a9408e1578f9761d9/recommender/commands.py#L68). Am I right?

File: tests/unit/engines/autoencoders/inference/test_embedding_component.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+def test_embedding_component(simulate_data_preparation_step, ae_pipeline_config):
+    ae_model_train_step = AEModelTrainingStep(ae_pipeline_config)
+
+    data_prep_step, _ = simulate_data_preparation_step
+    _, _ = ae_model_train_step(data_prep_step)
+
+    # Save Embedders
+    ae_model_train_step.save()
+
+    EmbeddingComponent()()
```
Comment (wujuu) [inline tests/unit/engines/autoencoders/inference/test_embedding_component.py:21]: We should add some assertions here, for example if the dense tensors in the DB are of a certain shape and maybe some more

**Review by wujuu — CHANGES_REQUESTED** [review]
Looks very good, some minor changes needed. We also need to separate other tests into unit and integration but that's a scope for another issue

**Review by wujuu — APPROVED** [review]
Awesome, just squash, rebase, resolve conflicts and we are ready to close this PR

**Pull request merged**